test/jasmine/tests/mapbox_test.js

```diff
@@ -1226,11 +1226,18 @@ describe('@noCI, mapbox plots', function() {
             expect(layout.zoom).toBeCloseTo(zoom);
         }
 
-        function _assert(center, zoom) {
+        function _assert(center, zoom, lonMin, latMin, lonMax, latMax) {
             _assertLayout(center, zoom);
 
             expect([evtData['mapbox.center'].lon, evtData['mapbox.center'].lat]).toBeCloseToArray(center);
             expect(evtData['mapbox.zoom']).toBeCloseTo(zoom);
+            expect(evtData['mapbox._derived']).toEqual({
+                coordinates: [
+                    [lonMin, latMax],
+                    [lonMax, latMax],
+                    [lonMax, latMin],
+                    [lonMin, latMin]
```

Comment:
Although mapbox for the moment does not support tiles in various coordinates system, I could imagine we may want to make this more general: https://github.com/archmoj/grib2class/blob/a4730169516fc532298385b0115d7a254e50d3f8/grib2class.js#L506-L536
So for latitude/longitude system I was wondering if we could return something like: 
```js
grid: {
  type: 'Latitude/longitude',
  lat1: 
  lat2:
  lon1:
  lon2:
}
```

Reply:
I went with the 2D array to match the expected form of the `coordinates` property in the mapbox image layer: https://docs.mapbox.com/mapbox-gl-js/example/image-on-a-map/

Note that the coordinates of all 4 corners of the viewport are required in general because in the presence of `bearing` and `pitch` the viewport can be a trapezoid.

That said, I'm not opposed to having some sort of `type` or `units` property in there if that makes sense to you.

Reply:
@jonmmease Thanks for the info. 
Now it make complete sense to me to keep it the way it is.
Would you be interested to adding tests to cover cases with pitch and bearing?
Also if you could possibly add one test for a scanrio where lon1 > lon2, that would be fantastic.

Reply:
From what I can tell, this logic will never produce a case where lon1 > lon2.  This is because mapbox will report longitude values less than zero and/or greater than 360 so that there is no discontinuity within the current viewport.  Because of that, I think the lonMin/lonMax names above are accurate.

For testing pitch and bearing, I suppose we could set up a test case that simulates a drag using the right mouse button. But I guess I don't personally see a lot of value in that since none of the plotly.js logic here depends on pitch or bearing. Do you see it differently?

Reply:
> From what I can tell, this logic will never produce a case where lon1 > lon2. This is because mapbox will report longitude values less than zero and/or greater than 360 so that there is no discontinuity within the current viewport. Because of that, I think the lonMin/lonMax names above are accurate.

I still think that by changing the `bearing` (e.g. to 180) one could get `lat1 > lat2` and `lon1 > lon2` at any location.
Am I missing something?

Reply:
Oh, good point. Changed names in 4b2623fda84e818e600c9539ea45b3ce80b558ba.